### PR TITLE
Ask query2

### PR DESCRIPTION
**Pull request by anuprajvarma**

### CODE REVIEW

File: src/pages/[creator_id]/ask-query.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+import Link from "next/link";
+import { useRouter } from "next/router";
+import React, { useState } from "react";
+import { type UseFormProps, useForm } from "react-hook-form";
+import { object, string, type z } from "zod";
+
+const contentLimit = 500;
+
+const askedQueryFormSchema = object({
+  question: string().max(contentLimit).nonempty("Please enter feedback."),
```
Comment (rosekamallove) [inline src/pages/[creator_id]/ask-query.tsx:21]: Feedback thodi mang rhe hai, non empty ka error message thk kro

File: src/pages/[creator_id]/ask-query.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  const { mutateAsync: addQueryMutation, isLoading: addQueryLoading } =
+    api.askedQuery.add.useMutation();
+
+  const [submitted, setSubmitted] = useState(false);
+
+  const ctx = api.useContext();
+
+  const { successToast, errorToast, warningToast } = useToast();
+
+  const revalidate = useRevalidateSSG();
```
Comment (rosekamallove) [inline src/pages/[creator_id]/ask-query.tsx:66]: Yeh use hora hai kya? Hata do ise

File: src/pages/[creator_id]/ask-query.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      ) : submitted ? (
+        <div className="flex w-full max-w-3xl flex-col items-center gap-2">
+          <h1 className="text-2xl md:text-3xl">
+            {"🙏"} Thank you for your query!
+          </h1>
+          <Link
+            className="text-medium group mt-8 flex items-center text-xl text-pink-500"
+            href={`/dashboard/queries`}
+          >
+            Watch your queries
```
Comment (rosekamallove) [inline src/pages/[creator_id]/ask-query.tsx:81]: "See your queries"

File: src/pages/[creator_id]/ask-query.tsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+              src={creator?.image ?? ""}
+              width={30}
+              height={30}
+              alt={creator?.name ?? ""}
+              className="rounded-full"
+            />
+            <span className="font-medium">{creator?.name ?? ""}</span>
+          </div>
+
+          <textarea
```
Comment (rosekamallove) [inline src/pages/[creator_id]/ask-query.tsx:133]: Change this to a markdown input, see event create to understand how to use this

File: src/pages/[creator_id]/index.tsx [hunk 5]
```diff
@@ -207,13 +211,13 @@ const Index = ({ creatorProfile }: CreatorPageProps) => {
...
-              href={`/${creator?.creatorProfile ?? ""}/query`}
-              className="group flex w-2/7 items-center justify-center gap-2 rounded-xl bg-neutral-200/10 px-4 py-2 pr-[1.2rem] text-sm font-medium text-neutral-300 backdrop-blur-sm transition-all duration-300 hover:bg-pink-500/80 hover:text-neutral-200"
+            <Link
+              id="query"
+              href={`/${creator?.creatorProfile ?? ""}/ask-query`}
+              className="w-2/7 group flex items-center justify-center gap-2 rounded-xl bg-neutral-200/10 px-4 py-2 pr-[1.2rem] text-sm font-medium text-neutral-300 backdrop-blur-sm transition-all duration-300 hover:bg-pink-500/80 hover:text-neutral-200"
             >
-              <FontAwesomeIcon icon={faQuestion} /> Ask any query to me
-            </Link> */}
+              <FontAwesomeIcon icon={faQuestion} /> Ask query
```
Comment (rosekamallove) [inline src/pages/[creator_id]/index.tsx:219]: "Ask a query"

File: src/pages/creator/dashboard/queries/index.tsx [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+    id: session?.user.id ?? "",
+  });
+
+  const { data: queries } = api.askedQuery.getAllCreator.useQuery({
+    creatorProfile: creator?.creatorProfile ?? "",
+  });
+
+  const [createReplyModal, setCreateReply] = useState<boolean>(false);
+  const [ReplyModal, setReplyModal] = useState<
+    (AskedQuery & { user: User }) | undefined
```
Comment (rosekamallove) [inline src/pages/creator/dashboard/queries/index.tsx:33]: Try not use this, if possible

File: src/pages/creator/dashboard/queries/index.tsx [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+          <div className="flex w-full flex-col items-center justify-center gap-2 p-4">
+            <div className="aspect-square object-contain">
+              <FontAwesomeIcon
+                icon={faCommentDots}
+                fontSize={200}
+                className="text-neutral-700"
+              />
+            </div>
+            <p className="mb-2 text-center text-sm text-neutral-400 sm:text-base">
+              You have not wrote any Queries.
```
Comment (rosekamallove) [inline src/pages/creator/dashboard/queries/index.tsx:115]: "You have not recieved any queries"

File: src/pages/creator/dashboard/queries/replied.tsx [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+import { usePathname } from "next/navigation";
+
+const RepliedAnswer = () => {
+  const { data: session } = useSession();
+
+  const { data: creator } = api.askedQuery.getAllCreatorProtected.useQuery({
+    id: session?.user.id ?? "",
+  });
+
+  const { data: queries } = api.askedQuery.getAllCreator.useQuery({
```
Comment (rosekamallove) [inline src/pages/creator/dashboard/queries/replied.tsx:21]: getAll dono jgh mat use kro make two separate functions

- getAllReplied
- getAllPending

File: src/server/api/routers/askedQuery.ts [hunk 9]
```diff
@@ -1,126 +1,116 @@
...
+export const askedQueryRouter = createTRPCRouter({
+  add: protectedProcedure
+    .input(z.object({ creatorProfile: z.string(), question: z.string() }))
+    .mutation(async ({ ctx, input }) => {
+      const { creatorProfile, question } = input;
+      const { prisma } = ctx;
+
+      if (!ctx.session.user.id) throw new TRPCError({ code: "UNAUTHORIZED" });
+
+      const creator = await prisma.user.findUnique({
```
Comment (rosekamallove) [inline src/server/api/routers/askedQuery.ts:19]: rename ths to `user`

File: src/server/api/routers/askedQuery.ts [hunk 10]
```diff
@@ -1,126 +1,116 @@
...
+          userId: ctx.session.user.id,
+          question,
+          creatorProfile,
+        },
+      });
+
+      return querys;
+    }),
+
+  update: protectedProcedure
```
Comment (rosekamallove) [inline src/server/api/routers/askedQuery.ts:41]: Update hata do

File: src/server/api/routers/askedQuery.ts [hunk 11]
```diff
@@ -1,126 +1,116 @@
...
+        },
+        include: {
+          user: true,
+        },
+      });
+
+      return querys;
+    }),
+
+  getAllCreatorProtected: protectedProcedure
```
Comment (rosekamallove) [inline src/server/api/routers/askedQuery.ts:100]: Don't use this, see `creator.ts` to get data with `session.user.id`

File: src/server/api/routers/askedQuery.ts [hunk 12]
```diff
@@ -1,126 +1,116 @@
...
+      const querys = await prisma.askedQuery.findFirst({
+        where: {
+          userId: ctx.session.user.id,
+          creatorProfile: input.creatorProfile,
+        },
+      });
+      return querys;
+    }),
+
+  getAllCreator: publicProcedure
```
Comment (rosekamallove) [inline src/server/api/routers/askedQuery.ts:83]: getQuery rakhko iska naam

**Review by rosekamallove — COMMENTED** [review]
Fix the changes mentioned.

**Review by curiousyuvi — APPROVED** [review]

**Pull request merged**